Assets/BossRoom/Scripts/Client/Game/State/ClientCharSelectState.cs

```diff
@@ -373,9 +385,50 @@ private void ConfigureUIForLobbyMode(LobbyMode mode)
         /// <summary>
         /// Called directly by UI elements!
         /// </summary>
+        /// <remarks>
+        /// For a reactive character select screen, the local player's selection is presented instantaneously
+        /// before a NetworkListEvent can be received from the server.
+        /// To accomplish this, the previous seat selected will be cleared or re-populated by a shared player, and
+        /// the new seat will be populated with the local selection.
+        /// </remarks>
         /// <param name="seatIdx"></param>
         public void OnPlayerClickedSeat(int seatIdx)
         {
+            if (m_LastSeatSelected != -1)
+            {
+                var otherPlayerSharesSeat = false;
+
+                // if any other user shares seat, set them as owner; else deactivate seat
+                foreach (var lobbyPlayer in CharSelectData.LobbyPlayers)
+                {
+                    if (lobbyPlayer.ClientId == NetworkManager.Singleton.LocalClientId)
+                    {
+                        // ignore self;
+                        continue;
+                    }
+
+                    if (lobbyPlayer.SeatIdx == m_LastSeatSelected)
+                    {
+                        // populate this seat with a shared player's lobby player state
+                        otherPlayerSharesSeat = true;
+                        m_PlayerSeats[m_LastSeatSelected].SetState(lobbyPlayer.SeatState, lobbyPlayer.PlayerNum, lobbyPlayer.PlayerName);
```

Comment:
If we're to show this in doc, let's rename m_PlayerSeats. I had to go to definition to figure out this was UI elements only (since it has a "SetState" method, I thought this was a netvar of some sort)

Reply:
We should maybe add a comment above saying this is the client driven part, where we just update our UI state without waiting for server values.

Reply:
Might be worth to note too that we're doing a pass over all seats. This serves as a sort of "merge conflict resolution" where instead of just trying to apply deltas, we're resetting everything.